internal/controller/multiclusterservice_controller.go

```diff
@@ -83,43 +84,37 @@ func (r *MultiClusterServiceReconciler) Reconcile(ctx context.Context, req ctrl.
 	return r.reconcileUpdate(ctx, mcs)
 }
 
-func (r *MultiClusterServiceReconciler) reconcileUpdate(ctx context.Context, mcs *kcm.MultiClusterService) (_ ctrl.Result, err error) {
-	if updated, err := utils.AddKCMComponentLabel(ctx, r.Client, mcs); updated || err != nil {
-		return ctrl.Result{Requeue: true}, err // generation has not changed, need explicit requeue
+func (*MultiClusterServiceReconciler) initServicesConditions(mcs *kcm.MultiClusterService) (changed bool) {
+	for _, typ := range [3]string{kcm.SveltosClusterProfileReadyCondition, kcm.FetchServicesStatusSuccessCondition, kcm.ServicesReferencesValidationCondition} {
+		if apimeta.SetStatusCondition(&mcs.Status.Conditions, metav1.Condition{
+			Type:               typ,
+			Status:             metav1.ConditionUnknown,
+			Reason:             kcm.ProgressingReason,
+			ObservedGeneration: mcs.Generation,
+		}) {
+			changed = true
+		}
```

Comment:
Why not just return?
```suggestion
		return apimeta.SetStatusCondition(&mcs.Status.Conditions, metav1.Condition{
			Type:               typ,
			Status:             metav1.ConditionUnknown,
			Reason:             kcm.ProgressingReason,
			ObservedGeneration: mcs.Generation,
		})
```

Reply:
We have to change all of the conditions, not just some of them

Reply:
Oh yeah, missed that.